File: src/172-factorial-trailing-zeroes/java/FactorialTrailingZeroes.java

public class FactorialTrailingZeroes {
    //iterative solution
    public int trailingZeroes(int n) {
        int ans = 0;
        while (n != 0) {
            int temp = n / 5;
            ans += temp;
            n = temp;
        }
        return ans;
    }
    //recursive one liner
    public int trailingZeroes(int n) {
        return n == 0 ? 0 : n / 5 + trailingZeroes(n / 5);
    }
}

File: src/172-factorial-trailing-zeroes/typescript/factorial-trailing-zeroes.ts
function trailingZeroes(n: number): number { //iterative
    let ans : number = 0;
    while (n != 0) {
        let temp : number = Math.floor(n / 5);
        ans += temp;
        n = temp;
    }
    return ans;
};

function trailingZeroes(n: number): number { //recursive
    if(n === 0) return 0;
    const tempNum : number = Math.floor(n / 5);
    return tempNum + trailingZeroes(tempNum);
};